editor/components/media-upload/check.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * WordPress dependencies
+ */
+import { compose, Fragment } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+
+export function MediaUploadCheck( { hasUploadPermissions, fallback, children } ) {
+	if ( ! hasUploadPermissions ) {
+		return fallback || null;
+	}
+
+	return <Fragment>{ children }</Fragment>;
+}
+
+export default compose( [
```

Comment:
Minor: With just a single higher-order component, you don't strictly need `compose`. Could be:

```js
export default withSelect( ( select ) =>  {
	// ...
} )( MediaUploadCheck );
```

... As you have it makes future refactoring easier though, so balances out; no need to change.